### PR TITLE
Integration2 t6f2 71 reviews controller

### PR DESCRIPTION
**Pull request by bovornc**
Finished working on controllers. Everything worked fine via Postman for me. Since our reviews are also Tuits, it can be a little awkward to test since we need to first create a Tuit, then create a Review using that Tuit's ID in the JSON body. 
Creating a review would have the JSON body format of:
{ 
"review": {"_id": [tuit id here]}, 
"restaurant": {"_id": [restaurantid]}
}

Updating a review to have a different tuit would just be putting 
{ 
"_id": [new tuit id here] 
} 
in the JSON body.

### CODE REVIEW

**Comment by bovornc** [comment]
### HTTP Endpoints:
findAllReviews: "/api/reviews"
findAllReviewsForRestaurant: "/api/restaurant/:restaurantid/reviews"
findAllReviewsByCritic: "/api/users/:criticid/reviews"
findReviewById: "/api/reviews/:reviewid"
createReview: "/api/restaurant/:restaurantid/users/:criticid/reviews"
updateReview: "/api/reviews/:reviewid"
deleteReview: "/api/reviews/:reviewid"

File: controllers/ReviewController.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+export default class ReviewController implements ReviewControllerI {
+    private static reviewController: ReviewController | null = null;
+    private static reviewDao: ReviewDao = ReviewDao.getInstance();
+
+    public static getInstance = (app: Express): ReviewController => {
+        if (ReviewController.reviewController === null) {
+            ReviewController.reviewController = new ReviewController();
+            app.get("/api/reviews", ReviewController.reviewController.findAllReviews);
+            app.get("/api/restaurant/:restaurantid/reviews", ReviewController.reviewController.findAllReviewsForRestaurant);
```
Comment (nala-he) [inline controllers/ReviewController.ts]: Hi Am, could you please change restaurant to plural "restaurants" for all the HTTP endpoints?

**Comment by nala-he** [comment]
I tested all functions on postman and they worked well. CreateReview and updateReview functions are a bit awkward as you have to create a tuit first and then link the tuit id and restaurant id. I am not sure whether it will be complicated to implement on the frontend side. I remembered that we agreed that a review is a type of tuit and this seems like the only way to implement it.  Would it be easier just to make review object independent from tuit? I guess we can discuss about this tomorrow during the meeting together.

**Comment by bovornc** [comment]
Changed HTTPs to "restaurants", and yeah I agree that it's rather clunky the way reviews are now.

**Comment by nala-he** [comment]
> Changed HTTPs to "restaurants", and yeah I agree that it's rather clunky the way reviews are now.

Thank you, Am! I guess we will approve once you finish refactoring the backend to make the review object standalone.

**Pull request closed without merging**